### PR TITLE
Add 500 check for registry api call

### PR DESCRIPTION
**Pull request by ankushagarwal**
Partially Addresses #14326

Signed-off-by: Ankush Agarwal ankushagarwal11@gmail.com

### CODE REVIEW

File: registry/auth.go [hunk 1]
```diff
@@ -180,6 +180,8 @@ func loginV1(authConfig *cliconfig.AuthConfig, registryEndpoint *Endpoint) (stri
 				}
 				// *TODO: Use registry configuration to determine what this says, if anything?
 				return "", fmt.Errorf("Login: Account is not Active. Please see the documentation of the registry %s for instructions how to activate it.", serverAddress)
+			} else if resp.StatusCode == 500 { // Issue #14326
+				return "", fmt.Errorf("Internal Server Error")
```
Comment (stevvooe) [inline registry/auth.go]: Please provide more information in a log output. Bowling over the 500 error with no context will make these impossible to debug at a later date.

Comment (ankushagarwal) [inline registry/auth.go]: I could, but I am not sure what else to add. The `body` is a huge html dump from the `index` service. I could update the Error message to something more appropriate like `index.docker.io returned a 500 while trying to authenticate`, is that what you're looking for?

Comment (stevvooe) [inline registry/auth.go]: @ankushagarwal It is okay to dump this to the logs. If we hadn't seen the body, we'd have no clue what the actual failure was. If you just say `index.docker.io returned a 500 while trying to authenticate`, the root cause will be impossible to track down.

**Comment by ankushagarwal** [comment]
@stevvooe : Updated and added log message

**Comment by stevvooe** [comment]
LGTM

PTAL @dmcgowan @tiborvass

**Comment by calavera** [comment]
LGTM

**Pull request merged**